### ISSUE TITLE
Parameters of Pascal function prototype not parsed correctly

### ISSUE THREAD

**Issue by t-rapp** [NaturalDocs/NaturalDocs#23]
When I use NaturalDocs 2.0.1 to generate documentation of a Delphi (Pascal) project the function arguments of a function are not parsed correctly. The second parameter is used as the type information of the first parameter, then the second parameter is repeated.

Example:
```Pascal
// Function: Execute
function Execute( const SqlStr : UnicodeString; Params : tParams;
   var ErrorMsg : UnicodeString ) : Boolean;
```
(see [DemoMain.pas](https://github.com/NaturalDocs/NaturalDocs/files/1932609/DemoMain.pas.txt) for the full test file)

The generated output looks like this:
![screenshot-1](https://user-images.githubusercontent.com/20061583/39059190-1f8c301e-44be-11e8-9466-7bf3c7ed5a9d.png)

**Comment by NaturalDocs** [comment]
I fixed the bug associated with this.  Not only that, I added better support for modifiers like const and var.  I wasn't aware you could put modifiers before the parameter names in Pascal, so now they format better.  Not only that, if you document the parameters in the body they'll be included in the extracted type.

![untitled-1](https://user-images.githubusercontent.com/23171507/39221820-345e16c8-4807-11e8-89b3-102864d3fb72.png)

The changes are available here.  They'll be included in the next full release.
http://www.naturaldocs.org/temp/Natural_Docs_temp10.zip

**Comment by NaturalDocs** [comment]
Question: I assume I have this right, that if you put "const LoginName, Password: UnicodeString" the const only applies to LoginName?  Or does it apply to Password as well?  I'm assuming you would need to put "const LoginName, const Password: UnicodeString" for it to apply to both.

**Comment by t-rapp** [comment]
Thanks, the modifiers work correctly now. The support for type indication in parameter lists is great!

Regarding the question: When writing "const LoginName, Password: UnicodeString" the const modifier applies to both, LoginName _and_ Password parameter. Repeating the const before the Password parameter name is a syntax error. For details see http://docwiki.embarcadero.com/RADStudio/Berlin/en/Parameters_(Delphi).

**Comment by NaturalDocs** [comment]
Modifiers fixed in this version.  I'll look into the Languages.txt changes.

http://www.naturaldocs.org/temp/Natural_Docs_temp11.zip

**Comment by NaturalDocs** [comment]
The modifier fixes and Languages.txt changes were included in 2.0.2.
